packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -253,6 +253,37 @@ class FileCollection extends DocumentCollection {
     return resp.links.related
   }
 
+  /**
+   * Checks if the file belongs to the parent's hierarchy.
+   *
+   * @param  {string}  childFileID  The file ID of the child
+   * @param  {string}  childDirID   The dirID of the child
+   * @param  {string}  childPath    The child path
+   * @param  {string}  parentID     The parent's document id
+   * @return {boolean}              Whether the file is a parent's child
+   */
+  async isChild(childFileID, childDirID, childPath, parentID) {
+    if (childFileID === parentID || childDirID === parentID) {
```

Comment:
It's ok to have this shortcut if we have the data. However it seems you are requesting the childDirId only to test this shortcut. Could it be optional ?

Reply:
Ok, so based on your previous proposition, `isChildOf(childId|doc, parentID|doc, { dirID } = {})` ?

Reply:
Ok for me if you think you will really benefit of this because you often know the parentId